README.md

```diff
@@ -131,15 +141,364 @@ class Admin < Dill::Role
+
+    widget(:todo_item, description).delete
+  end
+end
+```
+
+
+## Widgets Wait to Find an Element
+Widgets block, meaning they pause code execution, and check the page for that element until they find it or they reach Capybara's timeout limit. This is handy for dynamic UI tests.
+
+```ruby
+  roles.myrole.create_todo_item("Buy Milk")
+  # the item form is submitted to server and on response
+  # from the server, a new item is appended to the list.
+  # 'widget' waits until the item appears on the list,
+  # or Capybara's timeout limit is reached
+
+  expect(roles.myrole).to see :todo_item, "Buy Milk"
+```
+
+
+## Widget Root
+The **root** of a widget is the element itself, a bit like `this` is in Javascript. When a widget is declared with the widget macro, the root declaration is implicit and equal to the element with the css class in the definition.
+
+```ruby
+class TodoManager < Dill::Role
+  widget :todo_item, '.todo-item' do
+    def delete
+      root.find('a.delete').click
+    end
+  end
+end
+```
+
+
+## HTML Attributes
+We can access the id and classes of the widget as well.
+
+```ruby
+  # <div id="todo_form" class="todo-item right-aligned">
+  widget(:todo_item).id #=> "todo_form"
+  widget(:todo_item).classes #=> ["todo-item", "right-aligned"]
+```
+
+
+#Forms<a name="forms"></a>
+Forms inherit all the properties of widgets, and have some of their own.
+
+
+## Form Declaration
+```ruby
+  # with explicit class
+  widget :todo_form, '.todo-form', Dill::Form do
+  end
+
+  # with the form macro
+  form :todo_form, '.todo-form' do
+  end
+```
+
+
+## Form Fields
+Forms can have form-field widgets defined, as well as any regular sub-widgets. Form field widgets by default try to match their second argument with the text of a label, an input name or an input id.
+
+```ruby
+  form :form_with_everything, '.form-with-everything' do
+    # text field
+    text_field :request, 'request'
+
+    # select field
+    select :state, 'state'
+
+    # checkbox field
+    check_box :receive_email, 'receive_email'
+
+    # regular sub-widget
+    widget :hidden_field, '.hidden-field'
+
+    # currently no built-in form support for radio buttons.
+  end
+```
+
+
+#### Defining Fields with CSS Selectors
+If you'd rather use a CSS selector, you can do that by passing the second argument as an array, with the selector as the array's first element.
+
+```ruby
+  text_field :request, ['.request-field-class'],
+```
+
+
+#### Getting Field Values
+You can get the current value of any of these elements at any time by referencing them on the form object.
+
+```
+  widget(:form_with_everything).state #=> 'CO'
+```
+
+
+## Submitting a Form
+Dill will easily submit a form for you, via the UI, with either of these methods.
+
+```ruby
+  params = { request: 'Call me', state: 'CO', receive_email: true }
+
+  # the submit macro
+  submit :form_with_everything, params
+
+  # the submit widget method
+  widget(:form_with_everything).submit_with(params)
```

Comment:
It may also be worth mentioning `set`:

```ruby
set :form_with_everything, params

widget(:form_with_everything).set params # (I think)
```
This sets the values, but doesn't submit the form.